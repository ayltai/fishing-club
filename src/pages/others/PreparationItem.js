// @flow
'use strict';

import React from 'react';
import PropTypes from 'prop-types';
import muiThemeable from 'material-ui/styles/muiThemeable';
import Checkbox from 'material-ui/Checkbox';
import { ListItem } from 'material-ui/List';

const PreparationItem = (props : Object) => {
    return (
        <ListItem
            leftCheckbox={<Checkbox />}
            primaryText={props.name}
            secondaryText={props.description}
            secondaryTextLines={2} />
    );
};

PreparationItem.propTypes = {
    name        : PropTypes.oneOfType([ PropTypes.string.isRequired, PropTypes.element.isRequired ]),
    description : PropTypes.oneOfType([ PropTypes.string, PropTypes.element ])
};

export default muiThemeable()(PreparationItem);
