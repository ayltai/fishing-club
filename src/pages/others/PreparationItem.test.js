'use strict';

import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import PreparationItem from './PreparationItem';

it('renders correctly', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <BrowserRouter>
            <MuiThemeProvider>
                <PreparationItem
                    name="Cash"
                    description="You can buy more drinks there." />
            </MuiThemeProvider>
        </BrowserRouter>,
        div);

    expect(div.innerHTML).toMatchSnapshot();
});
