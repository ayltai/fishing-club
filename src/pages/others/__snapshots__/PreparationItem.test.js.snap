// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `

<div data-reactroot>
  <label style="color: rgba(0, 0, 0, 0.87); display: block; font-size: 16px; line-height: 16px; position: relative; margin-left: 0px; padding: 16px 16px 16px 72px; cursor: pointer;">
    <div style="cursor: pointer; position: absolute; overflow: visible; display: block; height: auto; width: 24px; top: 12px; left: 16px;">
      <!-- react-empty: 4 -->
      <input type="checkbox"
             style="position: absolute; cursor: inherit; pointer-events: all; opacity: 0; width: 100%; height: 100%; z-index: 2; left: 0px; box-sizing: border-box; padding: 0px; margin: 0px;"
      >
      <div style="display: flex; width: 100%; height: 100%;">
        <div style="float: left; position: relative; display: block; width: 24px; margin-right: 16px; margin-left: 0px; height: 24px;">
          <div>
            <svg viewbox="0 0 24 24"
                 style="display: inline-block; color: rgba(0, 0, 0, 0.87); fill: rgba(0, 0, 0, 0.87); height: 24px; width: 24px; position: absolute; opacity: 1;"
            >
              <path d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z">
              </path>
            </svg>
            <svg viewbox="0 0 24 24"
                 style="display: inline-block; color: rgba(0, 0, 0, 0.87); fill: #00bcd4; height: 24px; width: 24px; position: absolute; opacity: 0;"
            >
              <path d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z">
              </path>
            </svg>
          </div>
          <div>
          </div>
          <div style="position: absolute; overflow: hidden; height: 100%; width: 100%; top: 0px; left: 0px;">
          </div>
        </div>
      </div>
    </div>
    <div>
      Cash
    </div>
    <div style="font-size: 14px; line-height: 18px; height: 36px; margin: 4px 0px 0px 0px; color: rgba(0, 0, 0, 0.54); overflow: hidden; text-overflow: ellipsis; display: -webkit-box;">
      You can buy more drinks there.
    </div>
  </label>
</div>

`;
